FWCore/Framework/interface/EventSetup.h

```diff
@@ -142,7 +145,47 @@ class EventSetup
       //NOTE: the records are not owned
       std::map<eventsetup::EventSetupRecordKey, eventsetup::EventSetupRecord const *> recordMap_;
       eventsetup::EventSetupKnownRecordsSupplier const* knownRecords_;
-};
+  };
+
+  // Free functions to retrieve an object from the EventSetup.
+  // Will throw an exception if the record or  object are not found.
+
+  template <typename R, typename T>
+  T const& get(EventSetup const& setup) {
+    ESHandle<T> handle;
+    // throw if the record is not available
+    setup.get<R>().get(handle);
+    // throw if the handle is not valid
+    return * handle.product();
+  }
+
+  template <typename R, typename T>
```

Comment:
Could the following three be collapsed into just one fuction via
```C++
template <typename R, typename T, typename L>
T const& get(EventSetup const& setup, L&& label) {
   ESHandle<T> handle;
   //throw if the record is not available
   setup.get<R>().get(std::forward(label), handle);
   // throw if the handle is not valid
   return * handle.product();
}
```

Not a big deal, but it makes future maintenance easier (say in the case we move to a `std::string_view` as an allowed argument).